cmd/test_test.go

```diff
@@ -272,3 +276,134 @@ test_p {
 		t.Fatalf("didn't get expected %s error when inlined schema is present; got: %v", ast.TypeErr, err)
 	}
 }
+
+func TestWatchMode(t *testing.T) {
+
+	files := map[string]string{
+		"/policy.rego":      "package foo\n p := 1",
+		"/policy_test.rego": "package foo\n test_p { p == 1 }",
+	}
+
+	test.WithTempFS(files, func(root string) {
+		var buf bytes.Buffer
+
+		testParams.output = &buf
+		testParams.watch = true
+
+		done := make(chan struct{})
+		go func() {
+			opaTest([]string{root})
+			<-done
+		}()
+
+		time.Sleep(500 * time.Millisecond)
+
+		// update the test
+		f, _ := os.OpenFile(path.Join(root, "policy_test.rego"), os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+		_, err := f.WriteString("package foo\n test_p { p == 2 }")
+		if err != nil {
+			t.Fatal(err)
+		}
+		f.Close()
+
+		time.Sleep(500 * time.Millisecond)
+
+		// update policy so test passes
+		f, _ = os.OpenFile(path.Join(root, "policy.rego"), os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+		_, err = f.WriteString("package foo\n p := 2")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		f.Close()
+
+		time.Sleep(500 * time.Millisecond)
+
+		// add new policy and test
+		if err := os.WriteFile(path.Join(root, "policy2.rego"), []byte("package bar\n q := \"hello\""), 0644); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := os.WriteFile(path.Join(root, "policy2_test.rego"), []byte("package bar\n test_q { q == \"hello\" }"), 0644); err != nil {
+			t.Fatal(err)
+		}
+
+		time.Sleep(500 * time.Millisecond)
+
+		testParams.stopChan <- syscall.SIGINT
+		done <- struct{}{}
+
+		if !strings.Contains(buf.String(), "PASS: 1/1") {
+			t.Fatalf("Expected:%s in result but \n\nGot:\n\n%s\n\n", "PASS: 1/1", buf.String())
+		}
+
+		if !strings.Contains(buf.String(), "FAIL: 1/1") {
+			t.Fatalf("Expected:%s in result but \n\nGot:\n\n%s\n\n", "FAIL: 1/1", buf.String())
+		}
+
+		// verify result after new policy added
+		if !strings.Contains(buf.String(), "PASS: 2/2") {
```

Comment:
If we made the above change, we could assert these after each update, instead of at the end.